engine/src/object.cpp

```diff
@@ -5571,6 +5571,46 @@ static void MCDeletedObjectPoolDestroy(MCDeletedObjectPool *p_pool)
     MCMemoryDelete(p_pool);
 }
 
+static MCDeletedObjectPool* MCDeletedObjectPoolCleanup(MCDeletedObjectPool *p_pool)
+{
+	while (p_pool != nil && p_pool->defunct && p_pool->references == 0)
+	{
+		MCDeletedObjectPool *t_this_pool = p_pool;
+		p_pool = p_pool->parent;
+		MCDeletedObjectPoolDestroy(t_this_pool);
+	}
+	
+	return p_pool;
+}
+
+static MCDeletedObjectPool* MCDeletedObjectPoolRelease(MCDeletedObjectPool *p_pool)
```

Comment:
I suggest making this not return a value as it is only used in (top-level) contexts where it is not required (and where it is, we actually want a different pool).